_posts/2017-08-30-webpacker-3-0.markdown

```diff
@@ -0,0 +1,26 @@
+---
+layout: post
+title: 'Webpacker 3.0: No separate process needed, less config generated'
+categories: releases
+author: dhh
+published: true
+date: 2017-08-30 13:00:00 -06:00
+---
+
+We’ve just made it even easier to use [Webpack](https://webpack.github.io/) with Rails with this third major release of [Webpacker](https://github.com/rails/webpacker). The two big changes are that a separate process is no longer needed in development and that the vast majority of the config now lives in the [Webpacker NPM package](https://www.npmjs.com/package/@rails/webpacker), so your `config/` directory stays clean and updates are much easier.
+
+We nixed the need for the separate process by allowing Webpacker to compile on-demand in development as well as testing. We’ve done a lot of work to speed up this process, and for lots of apps, the performance will be more than fine. But if you have a huge app, or if you like live reloading or hot module replacements, you can still use the `bin/webpacker-dev-server`. Webpacker will automatically detect if this process is running and start serving packs from there rather than on-demand. Really slick.
+
+We also dramatically cut down on the amount of config boilerplate that’s generated in the Rails `config/` directory. All the standard stuff is now living inside the Webpacker NPM module, which makes upgrading so much easier. And you can still overwrite any of the defaults as you please. So it’s a big win all around.
+
+Additionally, we’ve moved all the compilation and clobber logic out of the Rake tasks and into [the Webpacker singleton instance](https://github.com/rails/webpacker/blob/master/lib/webpacker/instance.rb) itself. This makes it easier to use Webpacker in custom setups, like if you don’t use Yarn or have a different deployment strategy than just `assets:precompile`.
+
```

Comment:
Should we use permanent links like 

https://github.com/rails/webpacker/blob/29cce672a525c113ae75dfe31455a15890ebb5d0/lib/webpacker/instance.rb

https://github.com/rails/webpacker/blob/29cce672a525c113ae75dfe31455a15890ebb5d0/lib/webpacker/configuration.rb

https://github.com/rails/webpacker/blob/29cce672a525c113ae75dfe31455a15890ebb5d0/lib/webpacker/compiler.rb